packages/discussions/src/utils/posts.ts

```diff
@@ -0,0 +1,54 @@
+import { IGroup, IItem } from "@esri/arcgis-rest-portal";
+import { IDiscussionParams } from "../types";
+
+/**
+ * Utility that parses a discussion URI string into its component parts
+ *
+ * @export
+ * @param {string} discussion
+ * @return {*}  {IDiscussionParams}
+ */
+export function parseDiscussionURI(discussion: string): IDiscussionParams {
+  // NOTE: cannot use new URL(discussion) to parse uri bc difference in node and browser implementation
```

Comment:
It might help other developers to provide some examples here of types of Discussion URI. 
I recall you determined it was better to use a namespaced URI (arcgis://) rather than a Web URL .

Reply:
explicit parsing here makes sense because it's simple. 

if it became more complex or we allowed unique URI it might be worth considering a library for isomorphic URI parsing (e.g. [url-parse](https://www.npmjs.com/package/url-parse))

Reply:
Thank you! When I moved this method from the API repo to here I realized the difference in browser and node `URL` implementations, and came up short in finding a polyfill. I would much prefer that over relying on my limited regex ability